pkg/reconciler/revision/controller.go

```diff
@@ -107,7 +109,14 @@ func newControllerWithOptions(
 		transport = rt
 	}
 
-	resolver := newBackgroundResolver(logger, &digestResolver{client: kubeclient.Get(ctx), transport: transport}, impl.EnqueueKey)
+	userAgent := "knative (github.com/knative/serving)"
```

Comment:
const

Reply:
hmmm,
do we really want to annotate knative commitID in the UA?

Reply:
I'd rather see something like `knative.dev/serving@{commit}`. I could be swayed by precedent elsewhere (what do docker and containerd send?), but I dislike the parenthetical.  We might also want to expose the component `knative.dev/serving` as a variable so that it can be adjusted via ldflags (I could see Cloud Run wanting to distinguish product traffic from OSS traffic for example).

Reply:
> I'd rather see something like knative.dev/serving@{commit}

I can change the parenthetical "comment", but the `/` separates "product" from "token", neither of which can contain `@` or `/`. 

> what do docker and containerd send?
```
docker/19.03.1 go/go1.11.2 git-commit/74b1e89 kernel/4.19.112+ os/linux arch/amd64 UpstreamClient(Go-http-client/1.1)
```

```
containerd/1.2.8
```

> We might also want to expose the component knative.dev/serving as a variable so that it can be adjusted via ldflags (I could see Cloud Run wanting to distinguish product traffic from OSS traffic for example).

That's fine with me. I'm happy to add this somewhere if you point me in the right direction (or add a TODO?).

Reply:
@mattmoor what would you like to do here?